src/full/Agda/TypeChecking/Monad/Base.hs

```diff
@@ -1267,7 +1267,7 @@ data CheckedTarget = CheckedTarget (Maybe ProblemId)
 data TypeCheckingProblem
   = CheckExpr Comparison A.Expr Type
   | CheckArgs ExpandHidden Range [NamedArg A.Expr] Type Type (ArgsCheckState CheckedTarget -> TCM Term)
-  | CheckProjAppToKnownPrincipalArg Comparison A.Expr ProjOrigin (List1 QName) A.Args Type Int Term Type
+  | CheckProjAppToKnownPrincipalArg Comparison A.Expr ProjOrigin (List1 QName) A.Args Type Int Term Type (Args, Type)
```

Comment:
Stylistic remark: rather not put a tuple here, but give these two new args separately.
If you want to transport meaning with the tupling, I suggest to add a named type like.
```haskell
data <IntendedMeaning> = <IntendedMeaning>
  { <IM>Args :: Args -- ^ Haddockumentation for `Args`.
  , <IM>Type :: Type -- ^ Haddockumentation for `Type`.
  }

Reply:
Good idea! Yes, I did mean to convey that "these two things go together." I introduced a named type and comments as you suggested. Now people might even get an idea what these two things are and why they go together.